mobile/modules/tasks/controllers/tasksList.client.controller.es6

```diff
@@ -36,43 +37,43 @@ class TasksListController {
 
   sortListBy(type) {
     switch (type) {
-    case 'priority': {
-      this.filter.name         = 'priority';
-      this.filter.priorityAsc = !this.filter.priorityAsc;
-      this.tasks               = this.$filter('orderBy')(this.tasks, this.filter.priorityAsc ? 'priority' : '-priority');
-      this.setFiltersToLocalStorage();
-      break;
-    }
-    case 'time': {
-      this.filter.name     = 'time';
-      this.filter.timeAsc = !this.filter.timeAsc;
-
-      if (this.filter.timeAsc) {
-        this.tasks = TasksListController.bulbSortForEndTime(this.tasks);
-      } else {
-        this.tasks = TasksListController.bulbSortForEndTime(this.tasks).reverse();
+      case 'priority': {
+        this.filter.name         = 'priority';
+        this.filter.priorityAsc = !this.filter.priorityAsc;
+        this.filteredTasks               = this.$filter('orderBy')(this.filteredTasks, this.filter.priorityAsc ? 'priority' : '-priority');
+        this.setFiltersToLocalStorage();
+        break;
       }
-      this.setFiltersToLocalStorage();
-      break;
-    }
-    case 'isComplete': {
-      if (this.filter.isComplete) {
-        this.tasks = this.tasks.filter(el => el.isComplete);
-      } else {
-        this.tasks = this.tasksCopy.slice()
+      case 'time': {
+        this.filter.name     = 'time';
+        this.filter.timeAsc = !this.filter.timeAsc;
+  
+        if (this.filter.timeAsc) {
+          this.filteredTasks = TasksListController.bulbSortForEndTime(this.filteredTasks);
+        } else {
+          this.filteredTasks = TasksListController.bulbSortForEndTime(this.filteredTasks).reverse();
+        }
+        this.setFiltersToLocalStorage();
+        break;
+      }
+      case 'isComplete': {
+        if (this.filter.isComplete) {
+          this.filteredTasks = this.tasks.filter(el => el.isComplete);
+        } else {
+          this.filteredTasks = this.tasks;
+        }
+        this.setFiltersToLocalStorage();
+        break;
       }
-      this.setFiltersToLocalStorage();
-      break;
-    }
     }
   }
 
-
   find() {
     this.Tasks.query().$promise
         .then((resolved) => {
           this.tasks     = resolved;
-          this.tasksCopy = resolved.slice();
+          this.filteredTasks     = resolved;
```

Comment:
filtered tasks were created to give an ability to filter completed/incomplete tasks; this was done previously in taskController for desktop app